src/xb-silo.c

```diff
@@ -1311,8 +1310,8 @@ static void
 xb_silo_init (XbSilo *self)
 {
 	XbSiloPrivate *priv = GET_PRIVATE (self);
-	priv->file_monitors = g_hash_table_new_full (g_str_hash, g_str_equal,
-						     g_free, (GDestroyNotify) xb_silo_file_monitor_item_free);
+	priv->file_monitors = g_hash_table_new_full (g_file_hash, (GEqualFunc) g_file_equal,
+						     g_object_unref, (GDestroyNotify) xb_silo_file_monitor_item_free);
```

Comment:
Doesn't this need another (GDestroyNotify) to avoid a compile warning?

Reply:
Nope, `g_object_unref` has the same signature as `GDestroyNotify`. No compile warnings for me here

Reply:
My mistake, apologies.